### PR TITLE
Store length of string in string_ref object

### PR DESCRIPTION
**Pull request by adizaimi**
Store length of string in string_ref object, and use that length instead of calling strlen().

Signed-off-by: Adi Zaimi <azaimi@bloomberg.net>

### CODE REVIEW

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/d418f3e6/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>2/388 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>basic_rowlocks_generated [core dumped]<br>sc_timepart_logicalsc_generated</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/d418f3e6/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/d418f3e6/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2615'>Resubmit</a></p>

**Review by nirbhayc — DISMISSED** [review]
Ok to merge.

**Comment by akshatsikarwar** [comment]
Thank you! `string_ref_string` -> `string_ref_cstr` or some other variation (`string_ref_get` or `string_ref_get_cstr`).

File: util/string_ref.h [hunk 1]
```diff
@@ -44,7 +44,8 @@ struct string_ref * create_string_ref(const char *str);
 struct string_ref * get_ref(struct string_ref *ref);
 void put_ref(struct string_ref **ref);
 void transfer_ref(struct string_ref **from, struct string_ref **to);
-const char *get_string(struct string_ref *ref);
+const char *string_ref_string(struct string_ref *ref);
```
Comment (akshatsikarwar) [inline util/string_ref.h:47]: This header should declare `struct string_ref` first.

File: util/string_ref.c [hunk 2]
```diff
@@ -117,11 +119,15 @@ void transfer_ref(struct string_ref **from, struct string_ref **to)
     *from = NULL;
 }
 
-const char *get_string(struct string_ref *ref)
+inline const char *string_ref_string(struct string_ref *ref)
```
Comment (akshatsikarwar) [inline util/string_ref.c:122]: No `inline`. Same for `len` below

**Comment by akshatsikarwar** [comment]
Still getting used to new workflow.. If you make the suggested changes and push, the PR will lose the approval it currently has?

**Review by akshatsikarwar — APPROVED** [review]
LGTM

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/0cc00169/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>2/388 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>lostwrite [setup failure]<br>basic_rowlocks_generated [core dumped]</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/0cc00169/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/0cc00169/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2615'>Resubmit</a></p>

**Comment by adizaimi** [comment]
Thanks for the suggestions!

**Pull request merged**